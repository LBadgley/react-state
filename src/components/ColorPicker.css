.ColorPicker {
  button {
    border: 1em;
    padding: 1.5em;
    border-radius: 100%;
    height: 5em;
    width: 5em;
    text-align: center;
    align-content: center;

  }
  div {
    margin-left: 30px;
    border-radius: 100%;
    width: 100px;
    height: 100px;
    text-align: center;
    justify-content: center;
    
  }
  .red {
    background-color: rgb(192, 1, 1);
    color: white;
  }

  .yellow {
    background-color: rgb(252, 207, 27);
    color: white;
  }

  .blue {
    background-color: rgb(22, 22, 207);
    color: white;
  }
  
}
