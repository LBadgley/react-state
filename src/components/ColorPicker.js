import React from 'react';
import PropTypes from 'prop-types';
import styles from './ColorPicker.css';

function ColorPicker({ selectedColor }) {
  return (
    <section className={styles.ColorPicker}>
      <button className={styles.red} onClick={() =>selectedColor('red')}>red</button>
      <button className={styles.blue} onClick={() =>selectedColor('blue')}>blue</button>
      <button className={styles.yellow} onClick={() =>selectedColor('yellow')}>yellow</button>
    </section>
  );
}

ColorPicker.propTypes = {
  selectedColor: PropTypes.func.isRequired
};

export default ColorPicker;
