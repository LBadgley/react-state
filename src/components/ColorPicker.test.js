import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('color picker test', () => {
  it('renders the selected color', () => {
    const wrapper = shallow(<ColorPicker setSelectedColor = { jest.fn() } />);
    expect(wrapper).toMatchSnapshot();
  });
});
