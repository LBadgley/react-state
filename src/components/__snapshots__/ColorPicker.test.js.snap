// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color picker test renders the selected color 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    red
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    blue
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    yellow
  </button>
</section>
`;
